src/message.c

```diff
@@ -2178,6 +2197,130 @@ void qd_message_compose_4(qd_message_t *msg, qd_composed_field_t *field1, qd_com
 }
 
 
+void qd_message_compose_5(qd_message_t        *msg,
+                          qd_composed_field_t *headers,
+                          qd_buffer_list_t    *body,
+                          bool                 complete)
+{
+    qd_message_content_t *content         = MSG_CONTENT(msg);
+    qd_buffer_list_t     *headers_buffers = headers ? qd_compose_buffers(headers) : 0;
+
+    DEQ_INIT(content->buffers);
+    if (headers_buffers)
+        DEQ_APPEND(content->buffers, (*headers_buffers));
+
+    if (body) {
+        DEQ_APPEND(content->buffers, (*body));
+    }
+
+    content->receive_complete = complete;
+}
+
+
+int qd_message_extend(qd_message_t *msg, qd_buffer_list_t *buffers)
```

Comment:
Does this need to take into account Q2 threshold or Max Message Size functionality?